src/parse.ts

```diff
@@ -442,7 +454,7 @@ export class Parser extends EE implements Warner {
         chunk = Buffer.concat([this[BUFFER], chunk])
         this[BUFFER] = undefined
       }
-      if (chunk.length < gzipHeader.length) {
+      if (chunk.length < Math.max(gzipHeader.length, zstdHeader.length)) {
```

Comment:
The header length to test shoudl be set once, not calculated every time. Put something like this at the top level of this module:

```ts
const ZIP_HEADER_LEN = Math.max(gzipHeader.length, zstdHeader.length)
```

Then here:

```ts
if (chunk.length < ZIP_HEADER_LEN) {
```